### PR TITLE
Testing/sad paths

### PR DESCRIPTION
**Pull request by ameliaeiger**
#### What's this PR do?
Add testing/error handling for all Sleep methods

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ Promise.all([userProfileData, userActivityData, userSleepData, userHydrationData
     sleepDataHelper(data[2].sleepData);
     hydrationDataHelper(data[3].hydrationData);
   })
-  .catch((error) => alert("Opps something went wrong. Try again later."));
+  .catch((error) => alert("Oops something went wrong. Try again later."));
```
Comment (CorCanavan) [inline src/scripts.js:40]: Good "catch"!

Comment (CorCanavan) [inline src/scripts.js:40]: Good "catch"!

File: src/sleep.js [hunk 2]
```diff
@@ -11,36 +11,55 @@ class Sleep {
...
     let specifiedObject = this.allUserInstances.find(userObject => userObject.date === day);
+    if (!specifiedObject) {
+      return "Error"
+    } else {
     return specifiedObject;
+    }
   };
   returnObjectByDate(day, objectType) {
     let specifiedObject = this.findUserDataObjectByDate(day);
+    if (!day || !objectType || !specifiedObject || !specifiedObject[objectType]){
```
Comment (CorCanavan) [inline src/sleep.js:43]: Nicely done! Def helps me to see how others approach sad path testing.

Comment (CorCanavan) [inline src/sleep.js:43]: Nicely done! Def helps me to see how others approach sad path testing.

**Review by CorCanavan — APPROVED** [review]

**Review by megschuetz — APPROVED** [review]
This all looks really good for sad path testing.

**Pull request merged**